PyTest 8.2+ instantiates the test class with "runTest" causing an error:
> AttributeError: 'FSDPTest' object has no attribute 'runTest'. Did you mean: 'run_test'?

Fix using backport of https://github.com/pytorch/pytorch/issues/127517
author: Alexander Grund (TU Dresden)

diff --git a/torch/testing/_internal/common_distributed.py b/torch/testing/_internal/common_distributed.py
index 48c8e4dabbb..b1910b5122c 100644
--- a/torch/testing/_internal/common_distributed.py
+++ b/torch/testing/_internal/common_distributed.py
@@ -549,8 +549,14 @@ class MultiProcessTestCase(TestCase):
     # or run the underlying test function.
     def __init__(self, method_name: str = "runTest") -> None:
         super().__init__(method_name)
-        fn = getattr(self, method_name)
-        setattr(self, method_name, self.join_or_run(fn))
+        try:
+            fn = getattr(self, method_name)
+            setattr(self, method_name, self.join_or_run(fn))
+        except AttributeError as e:
+            if method_name != 'runTest':
+                # we allow instantiation with no explicit method name
+                # but not an *incorrect* or missing method name
+                raise ValueError(f"no such test method in {self.__class__}: {method_name}") from e
 
     def setUp(self) -> None:
         super().setUp()
@@ -964,8 +970,14 @@ class MultiThreadedTestCase(TestCase):
 
     def __init__(self, method_name: str = "runTest") -> None:
         super().__init__(method_name)
-        test_fn = getattr(self, method_name, None)
-        setattr(self, method_name, self.join_or_run(test_fn))
+        try:
+            fn = getattr(self, method_name)
+            setattr(self, method_name, self.join_or_run(fn))
+        except AttributeError as e:
+            if method_name != 'runTest':
+                # we allow instantiation with no explicit method name
+                # but not an *incorrect* or missing method name
+                raise ValueError(f"no such test method in {self.__class__}: {method_name}") from e
 
     def perThreadSetUp(self):
         # super().setUp()  # TestCase.setUp() calls torch.manual_seed()
